File: ABB-setup/dv-processing-rel_1.7/docs/source/assets/code_samples/c++/reading_data/sample2.cpp
#include <dv-processing/io/camera_capture.hpp>

int main() {
	// Open any camera
	dv::io::CameraCapture capture;

	// Print the camera name
	std::cout << "Opened [" << capture.getCameraName() << "] camera, it provides:" << std::endl;

	// Check whether event stream is available
	if (capture.isEventStreamAvailable()) {
		// Get the event stream resolution, the output is a std::optional, so the value() method is
		// used to get the actual resolution value
		const cv::Size resolution = capture.getEventResolution().value();

		// Print the event stream capability with resolution value
		std::cout << "* Events at " << resolution << " resolution" << std::endl;
	}

	// Check whether frame stream is available
	if (capture.isFrameStreamAvailable()) {
		// Get the frame stream resolution
		const cv::Size resolution = capture.getFrameResolution().value();

		// Print the frame stream capability with resolution value
		std::cout << "* Frames at " << resolution << " resolution" << std::endl;
	}

	// Check whether the IMU stream is available
	if (capture.isImuStreamAvailable()) {
		// Print the imu data stream capability
		std::cout << "* IMU measurements" << std::endl;
	}

	// Check whether the trigger stream is available
	if (capture.isTriggerStreamAvailable()) {
		// Print the trigger stream capability
		std::cout << "* Triggers" << std::endl;
	}

	return 0;
}

File: ABB-setup/dv-processing-rel_1.7/docs/source/assets/code_samples/python/reading_data/sample2.py
import dv_processing as dv

# Open any camera
capture = dv.io.CameraCapture()

# Print the camera name
print(f"Opened [{capture.getCameraName()}] camera, it provides:")

# Check whether event stream is available
if capture.isEventStreamAvailable():
    # Get the event stream resolution
    resolution = capture.getEventResolution()

    # Print the event stream capability with resolution value
    print(f"* Events at ({resolution.width}x{resolution.height}) resolution")

# Check whether frame stream is available
if capture.isFrameStreamAvailable():
    # Get the frame stream resolution
    resolution = capture.getFrameResolution()

    # Print the frame stream capability with resolution value
    print(f"* Frames at ({resolution.width}x{resolution.height}) resolution")

# Check whether the IMU stream is available
if capture.isImuStreamAvailable():
    # Print the imu data stream capability
    print("* IMU measurements")

# Check whether the trigger stream is available
if capture.isTriggerStreamAvailable():
    # Print the trigger stream capability
    print("* Triggers")
